File: frontend/src/types/field.ts
export interface XPathEntry {
  name: string;
  source: string;
  xpath: string;
  template?: string;
  output_element?: string;
  line?: number;
}

export interface DependencyRef {
  field_name: string;
  relationship: string;
  source_type: string;
  file_path?: string;
  line_number?: number;
}

export interface JavaReference {
  class_name: string;
  method_name?: string;
  finding_type: string;
  code_snippet?: string;
  file_path?: string;
  line_number?: number;
}

export interface FieldDetail {
  jurisdiction_id: string;
  field_name: string;
  header: string;
  asset_classes: string[];
  config_type: string;
  xslt_logic?: string;
  xslt_file?: string;
  xslt_line?: number;
  input_xpaths: XPathEntry[];
  dependencies: DependencyRef[];
  java_references: JavaReference[];
}

File: backend/app/schemas/field.py
from __future__ import annotations

from pydantic import BaseModel


class XPathEntry(BaseModel):
    name: str
    source: str
    xpath: str
    template: str | None = None
    output_element: str | None = None
    line: int | None = None


class DependencyRef(BaseModel):
    field_name: str
    relationship: str
    source_type: str
    file_path: str | None = None
    line_number: int | None = None


class JavaReference(BaseModel):
    class_name: str
    method_name: str | None = None
    finding_type: str
    code_snippet: str | None = None
    file_path: str | None = None
    line_number: int | None = None


class FieldDetail(BaseModel):
    jurisdiction_id: str
    field_name: str
    header: str
    asset_classes: list[str]
    config_type: str
    xslt_logic: str | None = None
    xslt_file: str | None = None
    xslt_line: int | None = None
    input_xpaths: list[XPathEntry] = []
    dependencies: list[DependencyRef] = []
    java_references: list[JavaReference] = []
